### PR TITLE
Extend tests on annotations in Generator.

### PR DESCRIPTION
**Pull request by hgaiser**
This PR extends the tests on annotations by also checking if they don't exceed the image shape.

### CODE REVIEW

File: keras_retinanet/preprocessing/generator.py [hunk 1]
```diff
@@ -75,27 +75,33 @@ def load_annotations(self, image_index):
         raise NotImplementedError('load_annotations method not implemented')
 
     def load_annotations_group(self, group):
-        annotations_group = [self.load_annotations(image_index) for image_index in group]
+        return [self.load_annotations(image_index) for image_index in group]
 
+    def check_images_annotations(self, image_group, annotations_group, group):
```
Comment (de-vri-es) [inline keras_retinanet/preprocessing/generator.py]: `check` implies that the function returns true/false. However, it filters the annotations. How about `filter_annotations` or similar?

File: tests/preprocessing/test_generator.py [hunk 2]
```diff
@@ -76,16 +76,19 @@ def test_simple_filter(self):
...
                 [  0,   0, 10, 10],
             ]),
         ]
 
         simple_generator = SimpleGenerator(input_annotations_group)
+        annotations_group = simple_generator.load_annotations_group(simple_generator.groups[0])
         # expect a UserWarning
         with pytest.warns(UserWarning):
-            annotations_group = simple_generator.load_annotations_group(simple_generator.groups[0])
+            image_group, annotations_group = simple_generator.check_images_annotations([input_image], annotations_group, simple_generator.groups[0])
```
Comment (de-vri-es) [inline tests/preprocessing/test_generator.py]: Should we consider `check_image_annotations` to be a public interface? Or is it a private/protected implementation detail and/or extension point for subclasses?

In the latter case we should probably not test it directly but test the public interface instead. Or at the very least test that the public interface also filters properly.

File: keras_retinanet/preprocessing/generator.py (hunk 1 above)
Comment (hgaiser) [reply keras_retinanet/preprocessing/generator.py]: Makes sense, I agree.

File: tests/preprocessing/test_generator.py (hunk 2 above)
Comment (hgaiser) [reply tests/preprocessing/test_generator.py]: How do you propose to do that though? The public interface is through the `next()` function, but that returns anchor targets. Should we test on those anchor targets?

Comment (de-vri-es) [reply tests/preprocessing/test_generator.py]: As discussed in person: What we could do is have a dataset with two annotations with different labels, where one of the annotations has to be discarded by the filter. Then let the generator generate two values and assert that both of them have the label of the valid annotation.

This tests at the very least that the filtering is done.

**Comment by hgaiser** [comment]
Made the requested changes, can you check?

**Pull request merged**